File: Python/p476.py
class Solution:
    def findComplement(self, num: int) -> int:
        mul = 1
        res = 0

        while num>1:
            if (num % 2 == 0):
                res = res + mul
            mul = mul * 2
            num >>= 1
        return res


s = Solution()        
res = s.findComplement(5)
print(res)

res = s.findComplement(1)
print(res)

res = s.findComplement(2)
print(res)

File: Go/p476.go
package main

import (
	"fmt"
)

func findComplement(num int) int {
	var tmp = num
	var mask = 1
	
	for (tmp > 0) {
		num = num ^ mask
		mask = mask << 1
		tmp = tmp >> 1
	}
	
	return num;
}

func main() {
	fmt.Println(findComplement(5))
	fmt.Println(findComplement(2))
	fmt.Println(findComplement(1))
}
